test/entities/swap.test.ts

```diff
@@ -0,0 +1,129 @@
+                        balance: String(
+                            parseUnits('100000', tokens[1].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[1].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[2].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[2].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[2].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+            weightedPoolFactory({
+                id: '0x177127622c4a00f3d409b75571e12cb3c8973d3c000000000000000000000011' as Hex,
+                tokens: [
+                    {
+                        address: tokens[2].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[2].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[2].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[3].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[3].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[3].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+        ];
+
+        test('amountIn to be maximally 1% more then expected', () => {
+            const amountOut = TokenAmount.fromRawAmount(tokens[3], 1000000n);
+            const swap = new Swap({
+                paths: [new PathWithAmount(tokens, pools, amountOut)],
+                swapKind: SwapKind.GivenOut,
+            });
+
+            const slippage = 1;
+            const limits = swap.limits(slippage, amountOut);
+
+            expect(limits[0]).toEqual(1010000n);
+        });
+
+        test('amountOut to be maximally 1% less then expected', () => {
+            const amountIn = TokenAmount.fromRawAmount(
+                tokens[0],
+                1000000000000000000n,
+            );
+            const swap = new Swap({
+                paths: [new PathWithAmount(tokens, pools, amountIn)],
+                swapKind: SwapKind.GivenIn,
+            });
+
+            const slippage = 1;
+            const limits = swap.limits(slippage, amountIn);
```

Comment:
Same here.
I believe you should pass the `expectedAmountOut` as input to the limits calculation.
One way of checking that is that the output should be in 6 decimals as the tokenOut is USDT for this test case.